content/docs/cucumber/configuration.md

```diff
@@ -15,36 +15,136 @@ polyglot:
 # Type Registry
 
 {{% block "java,kotlin" %}}
-Parameter types let you convert parameters from cucumber-expressions to objects. Data table and doc string string
-types let you convert data tables and doc strings to objects. Like step definitions type definitions are part of the
-glue. When placed on the glue path Cucumber will detect them automatically.
-
-For instance, the following class registers a `ParameterType` of type Color, and a `DataTableType` of type Person:
+Parameter types let you convert parameters from cucumber-expressions to objects. Data table and doc string string types let you convert data tables and doc strings to objects. Like step definitions, type definitions are part of the glue. When placed on the glue path Cucumber will detect them automatically.
+For example, the following class registers custom "Author" data table type, "List<Book>" docstring type and "Book" parameter type:
 
 ```java
-package com.example;
+package com.example.app;
 
 import io.cucumber.java.DataTableType;
+import io.cucumber.java.DocStringType;
 import io.cucumber.java.ParameterType;
+import io.cucumber.java.en.Given;
 
+import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Steps {
 
-import static java.util.Locale.ENGLISH;
+    @DataTableType
+    public Author authorEntryTransformer(Map<String, String> entry) {
+        return new Author(
+            entry.get("firstName"),
+            entry.get("lastName"),
+            entry.get("famousBook"));
+    }
 
-public class StepDefinitions {
+    @ParameterType(name = "book", value = ".*")
+    public Book bookTransformer(String bookName) {
+    	return new Book(bookName);
 
-    @ParameterType("red|blue|yellow")
-    public Color color(String color){
-        return new Color(color);
+    @DocStringType(contentType = "book_list")
+    public List<Book> defineBookList(String docstring) {
+        return Stream.of(docstring.split("\\s")).map(Book::new).collect(Collectors.toList());
```

Comment:
In our examples I think we should try to use common use cases. So json, xml, yaml, ect. It is okay to reference an third party library to handle the json parsing (e.g https://github.com/ralfstx/minimal-json or https://github.com/FasterXML/jackson-databind).